src/controllers/executionPlanWebviewController.ts

```diff
@@ -0,0 +1,157 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { ReactWebViewPanelController } from "./reactWebviewController";
+import * as ep from "../reactviews/pages/ExecutionPlan/executionPlanInterfaces";
+import { WebviewRoute } from "../sharedInterfaces/webviewRoutes";
+import * as LocalizedConstants from "../constants/localizedConstants";
+import { homedir } from "os";
+import { fileExists } from "../utils/utils";
+
+export class ExecutionPlanWebViewController extends ReactWebViewPanelController<
+  ep.ExecutionPlanWebViewState,
+  ep.ExecutionPlanReducers
+> {
+  constructor(
+    context: vscode.ExtensionContext,
+    private executionPlanService: ep.ExecutionPlanService,
+    private executionPlanContents: string,
+    // needs ts-ignore because linter doesn't recognize that fileName is being used in the call to super
+    // @ts-ignore
+    private xmlPlanFileName: string
+  ) {
+    super(
+      context,
+      `${xmlPlanFileName} ${LocalizedConstants.executionPlan}`,  // Sets the webview title
+      WebviewRoute.executionPlanDocument,
+      {},
+      vscode.ViewColumn.Active,
+      {
+        dark: vscode.Uri.joinPath(
+          context.extensionUri,
+          "media",
+          "executionPlan_inverse.svg"
+        ),
+        light: vscode.Uri.joinPath(
+          context.extensionUri,
+          "media",
+          "executionPlan.svg"
+        ),
+      }
+    );
+    this.state.isLoading = true;
+    this.initialize();
+  }
+
+  private async initialize() {
+    this.state.sqlPlanContent = this.executionPlanContents;
+	  this.state.theme = vscode.window.activeColorTheme.kind === vscode.ColorThemeKind.Dark ? "dark" : "light";
+    this.state.localizedConstants = LocalizedConstants;
+
+    await this.createExecutionPlanGraphs();
+    this.state.totalCost = this.calculateTotalCost();
+    this.registerRpcHandlers();
+    this.state.isLoading = false;
+  }
+
+  private registerRpcHandlers() {
+    this.registerReducer("getExecutionPlan", async (state, payload) => {
+      await this.createExecutionPlanGraphs();
+
+      return {
+        ...state,
+        executionPlan: this.state.executionPlan,
+        executionPlanGraphs: this.state.executionPlanGraphs
+      };
+    });
+    this.registerReducer("saveExecutionPlan", async (state, payload) => {
+      const homeDir = homedir();
+      const documentsFolder = vscode.Uri.file(`${homeDir}/Documents`);

```

Comment:
What about other OSes? Will this work for Mac and linux too? Also, can you please use join path instead. This might give some problems across different oses.

`vscode.Uri.joinPath `